### PR TITLE
feat(Cracked.io): add presence

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Added cracked.io presence
80% of the code was already written by Snowwy#1337, but they didn't know how to implement settings that's what I added and I am fixing the PR for simplicity sake.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/C/Cracked.io/presence.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+				presenceData.buttons = [
+					{
+						label: strings.buttonViewPage,
+						url: href,
+					},
+				];
+			}
+		}
+	}
+	// }
```
Comment (DarkVillager) [inline websites/C/Cracked.io/presence.ts:211]: Will remove this

File: websites/C/Cracked.io/presence.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+			viewCategory: "general.viewCategory",
+			viewHome: "general.viewHome",
+			buttonViewPage: "general.buttonViewPage",
+			buttonViewProfile: "buttonViewProfile",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+function capitalizeFirstLetter(string: string) {
+	if (string) {
```
Comment (EncryptedDev) [inline websites/C/Cracked.io/presence.ts:21]: Check isn't needed

File: websites/C/Cracked.io/presence.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+					document
+						.querySelector('[class="conv_users_avatar"]')
+						?.querySelector("img")
+						?.getAttribute("src") ?? Assets.Logo;
+			} else if (checkForChat)
+				presenceData.details = "Viewing private messages";
+			else if (active) {
+				presenceData.details = strings.viewHome;
+				presenceData.state =
+					active.getAttribute("title")?.toLowerCase().replace("home", "") !== ""
```
Comment (EncryptedDev) [inline websites/C/Cracked.io/presence.ts:95]: An empty string is falsy

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/C/Cracked.io/presence.ts (hunk 2 above)
Comment (DarkVillager) [reply websites/C/Cracked.io/presence.ts:21]: It is cuz it may be pass undefiend values.

File: websites/C/Cracked.io/presence.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+			buttonViewProfile: "buttonViewProfile",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+function capitalizeFirstLetter(string: string) {
+	if (!string) return "Undefined";
+	return (
+		string.trim().charAt(0).toUpperCase() + string.trim().slice(1).toLowerCase()
+	);
```
Comment (Bas950) [inline websites/C/Cracked.io/presence.ts:24]:
```suggestion
	return string.trim().charAt(0).toUpperCase() + string.trim().slice(1).toLowerCase();
```
cant it just be this?

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/C/Cracked.io/presence.ts (hunk 4 above)
Comment (DarkVillager) [reply websites/C/Cracked.io/presence.ts:24]: It sometimes gives an undefined value

**Comment by Bas950** [comment]
Please run prettier

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**